nemo/collections/nlp/models/language_modeling/megatron_t5_speechlm_model.py

```diff
@@ -659,9 +781,21 @@ def predict_step(self, batch: Any, batch_idx: int, dataloader_idx: int = 0) -> A
                     speech_mask=speech_mask,
                     inference=True,
                 )
-                output_tokens = output_logits.argmax(dim=2)  # (B,T,8)
-                output_tokens_curr_timestep = output_tokens[:, t]
+                # output_logits (B, T, V, 8)
+                output_logits_currtimestep = (
+                    output_logits[:, t, :, :].permute(0, 2, 1).contiguous().view(-1, 1024)
+                )  # (B*8, V)
+                temperature = 0.7  # 0.01 for greedy and 0.7 for sampling
```

Comment:
Can we make this an input instead of hardcoding?

Reply:
Done